editor/src/messages/tool/tool_messages/spline_tool.rs

```diff
@@ -320,7 +406,35 @@ impl Fsm for SplineToolFsmState {
 	}
 }
 
-fn update_spline(tool_data: &mut SplineToolData, show_preview: bool, responses: &mut VecDeque<Message>) {
+/// Return `true` only if new segment is inserted to connect two end points in the selected layer otherwise `false`.
+fn join_path(document: &DocumentMessageHandler, mouse_pos: DVec2, tool_data: &mut SplineToolData, responses: &mut VecDeque<Message>) -> bool {
+	let Some((endpoint, _)) = tool_data.points.last().map(|p| *p) else {
+		return false;
+	};
+	// use preview_point to get current dragging position.
+	let preview_point = tool_data.preview_point;
+	let selected_nodes = document.network_interface.selected_nodes(&[]).unwrap();
+	let selected_layers = selected_nodes.selected_layers(document.metadata());
+	// get the closest point to mouse position which is not preview_point or end_point.
+	let Some((layer, point, _)) = closest_point(document, mouse_pos, PATH_JOIN_THRESHOLD, selected_layers, |cp| {
+		preview_point.is_some_and(|pp| pp == cp) || cp == endpoint
+	}) else {
```

Comment:
Please put the `closest_point(...)` function call on its own line, with its return value set to a variable, for improved formatting.